smarts/core/utils/geometry.py

```diff
@@ -36,3 +41,47 @@ def buffered_shape(shape, width: float = 1.0) -> Polygon:
     elif not isinstance(ls, Polygon):
         raise RuntimeError("Shapely `object.buffer` behavior may have changed.")
     return ls
+
+
+def triangulate_polygon(polygon: Polygon):
+    return [
+        tri_face
+        for tri_face in triangulate(polygon)
+        if tri_face.centroid.within(polygon)
```

Comment:
This is from `sumo_road_network.py` which I  moved to `smarts/core/geometry.py` since it works for meshing `opendrive_road_network.py`. 
I believe the `tri_face.centroid.within(polygon)` check is added to ensure we don't include a triangle which lies outside the polygon which can happen if the polygon is non-convex.

Check this out:
https://stackoverflow.com/questions/65019170/how-do-you-triangulate-a-polygon-in-shapely

![image](https://user-images.githubusercontent.com/47334253/147777388-682a12b0-d5a5-450d-bfef-3adf399ef571.png)

Reply:
So then the following geometry would still exist:

![147777388-682a12b0-d5a5-450d-bfef-3adf399ef571](https://user-images.githubusercontent.com/8764693/147837032-a31a63dc-0a91-49f4-a62b-626fd42ba7ff.jpg)

Although, I could also now see that the non-convex point were moved up and right it could put the rightmost triangle centre out of shape. It seems like it could cause both missing and filled in geometry.